### PR TITLE
[#2446] Fix updating resources with invalid providers

### LINKED ISSUE #2446: Error when updating resource caused by wrong provided by
@Marcelinna - please fill in the details

1. Go to PC
2. Select one of the resources
3. Fill field _Resource Providers_ by the provider for which there was a synchronization error in MP  (e.g the logo is not correct, the data format is not correct), edit the Webpage (*)  field
4. Click submit in PC - Resource is submitted correctly in PC
6. Go to MP
7. See problem Resource isn't udpate in MP

When resource admin in PC try to add a provider for which there was a synchronization error in MP to field Resource Providers, updating the whole resource in MP fails.

Expected result: the service should be updated correctly, the external provider should not affect the updating of the service

### PR DESCRIPTION
**Pull request by goreck888**
This PR fixes adding providers to the resource
even if chosen provider has improper data
Fixes #2446

### CODE REVIEW

File: spec/features/backoffice/services_spec.rb [hunk 1]
```diff
@@ -206,6 +206,26 @@
...
+      select invalid_provider.name, from: "Providers"
+
+      click_on "Update Resource"
+
+      expect(page).to have_content("Resource updated correctly")
+
+      service.reload
+
+      expect(service.providers.map(&:id)).to eq([invalid_provider.id])
+    end
```
Comment (jswk) [inline spec/features/backoffice/services_spec.rb:227]: Actually, when updating through `record.update(params)` as is done in `backoffice/services_controller.rb`, the failing validation doesn't stop saving. So this test is unnecessary - it also doesn't fail when the `validate: false` is removed.
I would remove it.

**Review by jswk — CHANGES_REQUESTED** [review]
Correct one test and strip the other as described. Otherwise it's good to go.

**Comment by jswk** [comment]
Aah, and rebase onto the new master.

File: spec/features/backoffice/services_spec.rb (hunk 1 above)
Comment (goreck888) [reply spec/features/backoffice/services_spec.rb:227]: Ok.

**Review by jswk — APPROVED** [review]

File: app/controllers/backoffice/services_controller.rb [hunk 2]
```diff
@@ -163,7 +163,7 @@ def perform_update
     if Service::Update.new(@service, attributes).call
       update_logo_from_session!
       session.delete(preview_session_key)
-      redirect_to backoffice_service_path(@service), notice: "Service updated correctly"
+      redirect_to backoffice_service_path(@service), notice: "Resource updated correctly"
     else
       add_missing_nested_models
       render :edit, status: :bad_request
```
Comment (Marcelinna) [inline app/controllers/backoffice/services_controller.rb:169]: You can change to "New service created successfully"

**Review by Marcelinna — APPROVED** [review]

File: app/controllers/backoffice/services_controller.rb (hunk 2 above)
Comment (goreck888) [reply app/controllers/backoffice/services_controller.rb:169]: Done

**Review by kmarszalek — APPROVED** [review]

**Pull request merged**